util/rest/aes.go

```diff
@@ -0,0 +1,73 @@
+package rest
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+type AESDecoderEncoder struct {
+	cipherBlock cipher.Block
+}
+
+func NewAESDecoderEncoder(key []byte) (*AESDecoderEncoder, error) {
+	c, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+	return &AESDecoderEncoder{
+		cipherBlock: c,
+	}, nil
+}
+
+func (d *AESDecoderEncoder) DecodeBody(request *http.Request) (io.Reader, error) {
+	data, err := ioutil.ReadAll(request.Body)
```

Comment:
defer request.Close()

Reply:
From docs:
	// For server requests the Request Body is always non-nil
	// but will return EOF immediately when no body is present.
	// The Server will close the request body. The ServeHTTP
	// Handler does not need to.